pkg/conversions/gatewayroute_types_conversion.go

```diff
@@ -66,14 +106,117 @@ func Convert_CRD_HTTPGatewayRouteAction_To_SDK_HttpGatewayRouteAction(crdObj *ap
+
+	if crdObj.Rewrite.Prefix != nil {
+		Convert_CRD_HTTPGatewayRouteRewritePrefix_To_SDK_HttpGatewayRouteRewritePrefix(crdObj.Rewrite, sdkObj.Rewrite)
+	}
+
+	if crdObj.Rewrite.Hostname != nil {
+		Convert_CRD_HTTPGatewayRouteRewriteHostname_To_SDK_HttpGatewayRouteRewriteHostname(crdObj.Rewrite, sdkObj.Rewrite)
+	}
+
+	if crdObj.Rewrite.Path != nil {
+		Convert_CRD_HTTPGatewayRouteRewritePath_To_SDK_HttpGatewayRouteRewritePath(crdObj.Rewrite, sdkObj.Rewrite)
+	}
+}
+
+func Convert_CRD_HTTPGatewayRouteRewritePrefix_To_SDK_HttpGatewayRouteRewritePrefix(crdObj *appmesh.HTTPGatewayRouteRewrite, sdkObj *appmeshsdk.HttpGatewayRouteRewrite) {
+	sdkObj.Prefix = &appmeshsdk.HttpGatewayRoutePrefixRewrite{
+		DefaultPrefix: crdObj.Prefix.DefaultPrefix,
+		Prefix:        crdObj.Prefix.Prefix,
+	}
+}
+
+func Convert_CRD_HTTPGatewayRouteRewriteHostname_To_SDK_HttpGatewayRouteRewriteHostname(crdObj *appmesh.HTTPGatewayRouteRewrite, sdkOsdkObj *appmeshsdk.HttpGatewayRouteRewrite) {
+	// TODO: Need to sync with AppMesh API team
+	sdkOsdkObj.Hostname = &appmeshsdk.GatewayRouteHostnameRewrite{
+		DefaultTargetHostname: crdObj.Hostname.DefaultTargetHostname,
+		TargetBased:           &appmeshsdk.GatewayRouteTargetBasedHostnameRewrite{},
+	}
+}
+
+func Convert_CRD_HTTPGatewayRouteRewritePath_To_SDK_HttpGatewayRouteRewritePath(crdObj *appmesh.HTTPGatewayRouteRewrite, sdkObj *appmeshsdk.HttpGatewayRouteRewrite) {
+	sdkObj.Path = &appmeshsdk.HttpGatewayRoutePathRewrite{Exact: crdObj.Path.Exact}
+}
+
 func Convert_CRD_HTTPGatewayRouteMatch_To_SDK_HttpGatewayRouteMatch(crdObj *appmesh.HTTPGatewayRouteMatch, sdkObj *appmeshsdk.HttpGatewayRouteMatch) error {
 	sdkObj.Prefix = crdObj.Prefix
+	if crdObj.Hostname != nil {
+		sdkObj.Hostname = &appmeshsdk.GatewayRouteHostnameMatch{}
+		Convert_CRD_GatewayRouteHostnameMatch_To_SDK_GatewayRouteHostnameMatch(crdObj.Hostname, sdkObj.Hostname)
+	}
+	sdkObj.Method = crdObj.Method
+	if crdObj.Headers != nil && len(crdObj.Headers) != 0 {
+		if err := Convert_CRD_HTTPGatewayRouteHeaders_To_SDK_HttpGatewayRouteHeaders(crdObj, sdkObj); err != nil {
+			return err
+		}
+	}
+	if crdObj.Path != nil {
+		Convert_CRD_HTTPGatewayPath_To_SDK_HttpGatewayPath(crdObj, sdkObj)
+	}
+	if crdObj.QueryParameters != nil && len(crdObj.QueryParameters) != 0 {
+		Convert_CRD_HTTPGatewayRouteQueryParams_To_SDK_HttpGatewayRouteQueryParams(crdObj, sdkObj)
+	}
 	return nil
 }
 
+func Convert_CRD_HTTPGatewayRouteQueryParams_To_SDK_HttpGatewayRouteQueryParams(crdObj *appmesh.HTTPGatewayRouteMatch, sdkObj *appmeshsdk.HttpGatewayRouteMatch) {
+	length := len(crdObj.QueryParameters)
+	sdkQueryParams := make([]*appmeshsdk.HttpQueryParameter, 0, length)
+	for _, queryParam := range crdObj.QueryParameters {
+		sdkQueryParam := &appmeshsdk.HttpQueryParameter{}
+		sdkQueryParam.Name = queryParam.Name
+		if queryParam.Match != nil {
+			sdkQueryParam.Match = &appmeshsdk.QueryParameterMatch{Exact: queryParam.Match.Exact}
+		} else {
+			sdkQueryParam.Match = nil
```

Comment:
Do we need to explicitly set it to nil? Shouldn't that be the default value anyways..

Reply:
I updated at other places, will fix this as well